metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -417,3 +424,19 @@ def strip_ldap_info(input_clean: bytes) -> str:
     """Converts a b'uid=username,ou=Groups,dc=internal,dc=machines'
     format to username"""
     return input_clean.decode().split(",")[0].lstrip("uid=")
+
+
+def parse_groups(attrs: Dict[str, Any], filter_key: str) -> List[str]:
+    """Converts a list of LDAP groups to Datahub corpgroup strings"""
+    if filter_key in attrs:
+        return [
+            f"urn:li:corpGroup:{strip_ldap_group_cn(ldap_group)}"
+            for ldap_group in attrs[filter_key]
+        ]
+    return []
+
+
+def strip_ldap_group_cn(input_clean: bytes) -> str:
+    """Converts a b'cn=group_name,ou=Groups,dc=internal,dc=machines'
```

Comment:
Will the format of the group name always look like this?

Reply:
`memberOf` is an Active Directory specific field, and it has this format in AD (see `All direct members of specified group` example) https://social.technet.microsoft.com/wiki/contents/articles/5392.active-directory-ldap-syntax-filters.aspx#Examples . There is no such field in other LDAP implementations as far as I know.

Reply:
If this is AD specific I wonder if we need any additional configurability for other LDAP distros?

Reply:
I don't think so, because if field does not exist ingester just don't add aspect to user, i.e. work as before. I found that this field could be enabled for OpenLDAP with overlay (see test data https://github.com/datahub-project/datahub/pull/5470/files#diff-2adbfea9193e41527a7594ee81e6000cc5df0336cb06e90cfc1af71c106dc72aR12) -  it also has such format.

Reply:
Okay thanks for clarifying!